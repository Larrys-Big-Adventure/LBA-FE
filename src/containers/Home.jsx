import React, { useState, useEffect } from 'react';
import styled from 'styled-components';

import axiosWithAuth from '../utils/axiosWithAuth';
import { Navigation, GameView, Controller, ActionView } from '../components';

const Home = () => {
  const [state, setState] = useState({
    id: null,
    name: '',
    error: '',
    title: '',
    loot: [],
    players: [],
    inventory: [],
    error_msg: '',
    description: '',
    isInventoryModalOpen: false,
    isLootModalOpen: false,
    isPlayersModalOpen: false
  });

  // init
  useEffect(() => {
    axiosWithAuth()
      .get('api/adv/init/')
      .then(res => {
        setState({
          ...state,
          id: res.data.uuid,
          name: res.data.name,
          title: res.data.title,
          players: res.data.players,
          description: res.data.description
        });
      })
      .catch(() => {
        setState({
          ...state,
          error: ''
        });
      });
    // eslint-disable-next-line
  }, []);

  // move
  const moveDirection = direction => {
    axiosWithAuth()
      .post('api/adv/move/', { direction })
      .then(res => {
        setState({
          ...state,
          name: res.data.name,
          title: res.data.title,
          players: res.data.players,
          error_msg: res.data.error_msg,
          description: res.data.description
        });
      })
      .catch(() => {
        setState({
          ...state,
          error: ''
        });
      });
  };

  const moveHandler = e => {
    if (e.key === 'w') {
      moveDirection('n');
    } else if (e.key === 'a') {
      moveDirection('w');
    } else if (e.key === 's') {
      moveDirection('s');
    } else if (e.key === 'd') {
      moveDirection('e');
    } else if (e.key === 'l') {
      setState({
        ...state,
        isInventoryModalOpen: false,
        isLootModalOpen: !state.isLootModalOpen,
        isPlayersModalOpen: false
      });
    } else if (e.key === 'i') {
      setState({
        ...state,
        isInventoryModalOpen: !state.isInventoryModalOpen,
        isLootModalOpen: false,
        isPlayersModalOpen: false
      });
    } else if (e.key === 'p') {
      setState({
        ...state,
        isInventoryModalOpen: false,
        isLootModalOpen: false,
        isPlayersModalOpen: !state.isPlayersModalOpen
      });
    }
  };

  console.log(state.title);
  console.log(state.description);
  return (
    <HomeContainer>
      <Navigation />
      <GameView
        moveDirection={moveDirection}
        moveHandler={moveHandler}
        loot={state.loot}
        players={state.players}
        inventory={state.inventory}
        isInventoryModalOpen={state.isInventoryModalOpen}
        isLootModalOpen={state.isLootModalOpen}
        isPlayersModalOpen={state.isPlayersModalOpen}
      />
      <BottomSection>
        <ActionView />
        <Controller moveDirection={moveDirection} moveHandler={moveHandler} />
      </BottomSection>
    </HomeContainer>
  );
};

const HomeContainer = styled.div`
  width: 100%;
  height: 100vh;
  display: flex;
  flex-direction: column;
`;

const BottomSection = styled.div`
  width: 100%;
  background-color: white;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  position: relative;
  bottom: 0;
`;

export default Home;

// HOME PAGE TO DO
// import game view
// import controllers
