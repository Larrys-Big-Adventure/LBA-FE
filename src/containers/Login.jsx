import React from 'react';
import styled from 'styled-components';
import { LoginForm } from '../components';

const Login = () => {
  return (
    <LoginContainer>
      <LoginForm />
    </LoginContainer>
  );
};

const LoginContainer = styled.div`
  margin: 0 auto;
`;

export default Login;
