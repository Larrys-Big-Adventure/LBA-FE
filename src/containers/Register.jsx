import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { RegisterForm, GameTitle } from '../components';
import {
  AuthTitle,
  BottomMessage,
  AuthCardContainer
} from '../styles/AuthCard';

import { Container } from 'nes-react';

const Register = ({ history }) => {
  return (
    <AuthCardContainer>
      <Container>
        <Container>Larry's Leisurely Adventure Game</Container>

        <Container>Sign up</Container>
        <RegisterForm history={history} />
        <Container>
          <link
            href="https://fonts.googleapis.com/css?family=Press+Start+2P"
            rel="stylesheet"
          />
          Already have an account?
          <Link to="/login"> Log in</Link>
        </Container>
      </Container>
    </AuthCardContainer>
  );
};

Register.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired
};

export default Register;
