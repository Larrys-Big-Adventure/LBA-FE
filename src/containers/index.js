import Home from './Home';
import Login from './Login';
import NoMatch from './NoMatch';
import Register from './Register';

export { Home, Login, NoMatch, Register };
