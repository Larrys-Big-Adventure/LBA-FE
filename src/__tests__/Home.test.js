import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { cleanup } from '@testing-library/react';
import render from '../__testHelpers__/render';

import { Home } from '../containers';

afterEach(cleanup);

describe('Home', () => {
  it('renders without crashing', () => {
    render(<Home />);
  });
});
