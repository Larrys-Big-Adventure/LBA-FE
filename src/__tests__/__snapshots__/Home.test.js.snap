// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches previous snapshot 1`] = `
<div>
  <h1>
    Home here
  </h1>
</div>
`;
